src/main/java/org/opensearch/knn/index/codec/KNN990Codec/KNN990QuantizationStateWriter.java

```diff
@@ -27,6 +27,10 @@ public final class KNN990QuantizationStateWriter {
     private List<FieldQuantizationState> fieldQuantizationStates = new ArrayList<>();
     static final String NATIVE_ENGINES_990_KNN_VECTORS_FORMAT_QS_DATA = "NativeEngines990KnnVectorsFormatQSData";
 
+    public static KNN990QuantizationStateWriter of(SegmentWriteState segmentWriteState) throws IOException {
```

Comment:
why we are choosing this over not creating a constructor?

Reply:
If thats the case lets annotate the function with visiblefortesting annotation and make it package private